### PR TITLE
chore: hook ProfilerAgent into harness [DET-5204]

### PR DESCRIPTION
**Pull request by armandmcqueen**
## Description

Hook the ProfilerAgent into the harness. 

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: harness/determined/profiler.py [hunk 1]
```diff
@@ -45,6 +45,15 @@ class ShutdownMessage:
     pass
 
 
+DEBUG = True
+
+
+def debug_log(*args: Any) -> None:
+    args_as_str = " ".join([str(arg) for arg in args])
+    if DEBUG:
+        logging.info(f"{LOG_NAMESPACE} (DEBUG) {args_as_str}")
```
Comment (shiyuann) [inline harness/determined/profiler.py:54]: blocking: This is a bit weird. Don't harness set the logger level to debug if it's in debug mode? why do you use `logging.info` here?

File: harness/determined/keras/callbacks.py [hunk 2]
```diff
@@ -701,3 +701,14 @@ def on_train_workload_begin(
...
+
+
+class _DeterminedProfiler(Callback):
+    """Hooks Keras into the profiler agent lifecycle"""
+
+    def __init__(self, prof: profiler.ProfilerAgent) -> None:
+        super().__init__()
+        self.prof = prof
+
+    def on_train_batch_begin(self, batch: int, logs: Optional[Dict] = None) -> None:
```
Comment (shiyuann) [inline harness/determined/keras/callbacks.py:713]: q: why is this logs argument not used and included?

File: harness/determined/profiler.py [hunk 3]
```diff
@@ -201,37 +232,74 @@ def _begin_collection(self) -> None:
...
+                debug_log("ProfilerAgent._end_collection - sysmetriccollector shutdown signal sent")
+                self.sys_metric_collector_thread.join()
+                debug_log("ProfilerAgent._end_collection - sysmetriccollector joined")
+
+            # TODO [DET-5062]: Shut down TimingBatcher as well
+            debug_log("ProfilerAgent._end_collection - shutting down sender thread")
+            self.sender_thread.send_shutdown_signal()
+            debug_log("ProfilerAgent._end_collection - sender thread shutdown signal sent")
+            self.sender_thread.join()
+            debug_log("ProfilerAgent._end_collection - sender thread joined")
```
Comment (rb-determined-ai) [inline harness/determined/profiler.py:275]: Are all these logs really necessary?

**Comment by armandmcqueen** [comment]
Closing in favor of #2355 to avoid rebase pain

**Pull request closed without merging**